cpp/include/cudf/lists/combine.hpp

```diff
@@ -49,9 +49,8 @@ enum class concatenate_null_policy { IGNORE, NULLIFY_OUTPUT_ROW };
  * r is now [{0, 1, 8}, {2, 3, 4, 9}, {5}, {10, 11, 12}, {6, 7, 13, 14, 15, 16}]
  * @endcode
  *
- * @throws cudf::logic_error if any column of the input table is not a lists columns.
- * @throws cudf::logic_error if any lists column contains nested typed entry.
- * @throws cudf::logic_error if all lists columns do not have the same entry type.
+ * @throws cudf::logic_error if any column of the input table is not a lists column.
+ * @throws cudf::logic_error if all lists columns do not have the same type.
```

Comment:
Same type may mean just their chidlren types (like lists-of-int <==> lists-of-int). If we have something like lists-of-lists and lists-of-lists, we still have the same type but one column may have many nested level.

```suggestion
 * @throws cudf::logic_error if all lists columns do not have the same structure.
```

Reply:
I think type is the correct term here.  For nested types, "type" and structure basically mean the same thing, but cudf lean towards using type as the terminology.  See:

https://github.com/rapidsai/cudf/blob/d1a019192695c31ab28a7d0702cfb6b0f84c7cbc/cpp/src/utilities/type_checks.cpp#L66